src/main/java/com/conveyal/datatools/manager/jobs/AutoDeployJob.java

```diff
@@ -0,0 +1,221 @@
+                deployment.name,
+                project.name
+            );
+            LOG.warn(message);
+            NotifyUsersForSubscriptionJob.createNotification(
+                "deployment-updated",
+                project.id,
+                message
+            );
+            return;
+        }
+
+        try {
+            synchronized (lockedProjects) {
+                if (!lockedProjects.contains(project.id)) {
+                    lockedProjects.add(project.id);
+                    LOG.info("Auto deploy lock added for project id: {}", project.id);
+                } else {
+                    LOG.warn("Unable to acquire lock for project {}", project.name);
+                    return;
+                }
+            }
+            LOG.info("Auto deploy task running for project {}", project.name);
+
+            // Get the most recently used server.
+            String latestServerId = deployment.latest().serverId;
+            OtpServer server = Persistence.servers.getById(latestServerId);
+            if (server == null) {
+                LOG.warn(
+                    "Server with id {} no longer exists. Skipping deployment for project {}.",
+                    latestServerId,
+                    project.name
+                );
+                return;
+            }
+
+            // Analyze and update feed versions in deployment.
+            Collection<String> updatedFeedVersionIds = new LinkedList<>();
+            List<FeedVersion> latestVersionsWithCriticalErrors = new LinkedList<>();
+            boolean shouldWaitForNewFeedVersions = false;
+
+            // Production ready feed versions.
+            List<Deployment.SummarizedFeedVersion> pinnedFeedVersions = deployment.retrievePinnedFeedVersions();
+
+            // Iterate through each feed version for deployment.
+            for (
+                Deployment.SummarizedFeedVersion currentDeploymentFeedVersion : deployment.retrieveFeedVersions()
+            ) {
+                // Retrieve the latest feed version associated with the feed source of the current
+                // feed version set for the deployment.
+                FeedVersion latestFeedVersion = currentDeploymentFeedVersion.feedSource.retrieveLatest();
+                // Make sure the latest feed version is not going to supersede a pinned feed version.
+                if (pinnedFeedVersions
+                    .stream()
+                    .anyMatch(feedVersion -> feedVersion.feedSource.id.equals(latestFeedVersion.feedSourceId))
+                ) {
+                    String message = String.format("Feed version %s from feed source %s could not be advanced because a previous feed version has been pinned for project %s.",
+                        latestFeedVersion.id,
+                        latestFeedVersion.parentFeedSource().name,
+                        project.name);
+                    LOG.warn(message);
```

Comment:
Probably not even worth logging a warning here since this is expected behavior.

Reply:
ok, have removed.